### PR TITLE
#219 Provide ability to define StorageClass for Databases

### PR DESCRIPTION
**Pull request by drichtarik**
Not sure about constants.go, defaults.go or resource/types.go edits
deyployments.go has comment section that I saved if it proves to be somewhat right approach

### CODE REVIEW

File: pkg/controller/quayecosystem/constants/constants.go [hunk 1]
```diff
@@ -112,6 +112,7 @@ const (
 	QuayRegistryStoragePath = "/datastorage/registry"
 	// QuayRegistryStoragePersistentVolumeStoreSize represents the size of the PersistentVolume that should be used for registry storage
 	QuayRegistryStoragePersistentVolumeStoreSize = "20Gi"
+	PersistentVolumeStorageClassName = "storageClassName"
```
Comment (sabre1041) [inline pkg/controller/quayecosystem/constants/constants.go]: No need for this constant value as no default will be used

File: pkg/controller/quayecosystem/provisioning/provision.go [hunk 2]
```diff
@@ -429,7 +429,7 @@ func (r *ReconcileQuayEcosystemConfiguration) createClairDatabase(meta metav1.Ob
 
 	// Create PVC
 	if !utils.IsZeroOfUnderlyingType(r.quayConfiguration.QuayEcosystem.Spec.Clair.Database.VolumeSize) {
-		databasePvc := resources.GetDatabasePVCDefinition(meta, r.quayConfiguration.QuayEcosystem.Spec.Clair.Database.VolumeSize)
+		databasePvc := resources.GetDatabasePVCDefinition(meta, r.quayConfiguration.QuayEcosystem.Spec.Clair.Database.VolumeSize, &r.quayConfiguration.QuayEcosystem.Spec.Quay.Database.StorageClass)
```
Comment (sabre1041) [inline pkg/controller/quayecosystem/provisioning/provision.go]: Should be Clair's database reference and not Quay

File: pkg/controller/quayecosystem/resources/deployments.go [hunk 3]
```diff
@@ -710,6 +710,15 @@ func GetDatabaseDeploymentDefinition(meta metav1.ObjectMeta, quayConfiguration *
 		})
 	}
 
+	/*	if !utils.IsZeroOfUnderlyingType(database.StorageClass) {
```
Comment (sabre1041) [inline pkg/controller/quayecosystem/resources/deployments.go]: Can probably remove this snippet

File: pkg/controller/quayecosystem/resources/types.go [hunk 4]
```diff
@@ -72,6 +72,7 @@ type DatabaseConfig struct {
 	Memory              string
 	VolumeSize          string
 	CredentialsName     string
+	StorageClass        string
```
Comment (sabre1041) [inline pkg/controller/quayecosystem/resources/types.go]: This value is not needed

File: pkg/controller/quayecosystem/resources/storage.go [hunk 5]
```diff
@@ -50,6 +50,7 @@ func GetDatabasePVCDefinition(meta metav1.ObjectMeta, volumeSize string) *corev1
 					corev1.ResourceName(corev1.ResourceStorage): resource.MustParse(volumeSize),
 				},
 			},
+			StorageClassName: storageClass,
 		},
```
Comment (sabre1041) [inline pkg/controller/quayecosystem/resources/storage.go:54]: Take a look at the method above for how the StorageClass is being added to the object. It should only be added if the value is defined and is not an empty string

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Looking good.  Made a few recommendations on some changes for the next draft iteration

**Comment by drichtarik** [comment]
Thank you for the input and review. Updated. Hopefully I didn't miss anything.

**Comment by sabre1041** [comment]
@drichtarik this looks very good!

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Can you generate the CRD's (4.x and 3.x) along with the generated classes + openapi specs. otherwise we should be good!

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**